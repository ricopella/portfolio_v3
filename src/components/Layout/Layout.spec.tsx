import createFixtureAuthorMetaData from '../../fixtures/createFixtureAuthorMetaData'
import React from 'react'
import renderer from 'react-test-renderer'
import { PureLayout as Layout } from './index'
import { render } from '@testing-library/react'

describe('<Layout />', () => {
  const props = {
    author: createFixtureAuthorMetaData(),
  }
  it('should match Layout snapshot', () => {
    const internalTree = renderer
      .create(
        <Layout {...props}>
          <div>Wu-Tang Clan</div>
        </Layout>
      )
      .toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render children', () => {
    const { getByText } = render(
      <Layout {...props}>
        <div>Wu-Tang Clan</div>
      </Layout>
    )

    expect(getByText(/Tom Brady/i)).toBeTruthy()
    expect(getByText(/Wu-Tang Clan/i)).toBeTruthy()
  })
})
