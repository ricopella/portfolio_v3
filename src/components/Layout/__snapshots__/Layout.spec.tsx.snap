// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match Layout snapshot 1`] = `
<div
  className="css-9vt5au"
>
  <div
    className="css-1c7fy2u"
  >
    <div>
      Wu-Tang Clan
    </div>
  </div>
  <footer
    className="css-1cda30e"
  >
    Â© 
    2020
    , 
    <span>
      Tom Brady
    </span>
  </footer>
</div>
`;
